actions/plugins/version-bump-changelog/action.yml

```diff
@@ -0,0 +1,115 @@
+
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      with:
+        token: ${{ steps.generate-github-token.outputs.token }}
+        persist-credentials: true
+        fetch-depth: 0
+
+    - name: Setup Git
+      shell: bash
+      run: |
+        git config user.name 'grafana-plugins-platform-bot[bot]'
+        git config user.email '144369747+grafana-plugins-platform-bot[bot]@users.noreply.github.com'
+
+    - name: Get previous tag before bumping
+      id: previous-tag
+      if: ${{ inputs.generate-changelog == 'true' }}
+      shell: bash
+      run: |
+        PREVIOUS_TAG=$(git describe --tags --abbrev=0 2>/dev/null || echo "v0.0.0")
+        echo "previous-tag=${PREVIOUS_TAG}" >> $GITHUB_OUTPUT
+
+    - name: Bump version
+      id: bump
+      shell: bash
+      run: |
+        NEW_VERSION=$(npm version ${INPUT_VERSION})
+        echo "new-version=${NEW_VERSION}" >> $GITHUB_OUTPUT
+      env:
+        INPUT_VERSION: ${{ inputs.version }}
+        GITHUB_TOKEN: ${{ steps.generate-github-token.outputs.token }}
+    
+    - name: Generate changelog
+      if: ${{ inputs.generate-changelog == 'true' }}
+      shell: bash
+      run: |
+        npm install -g generate-changelog
+
+        # Generate changelog into a temp file
+        if [[ "${{ steps.previous-tag.outputs.previous-tag }}" == "v0.0.0" ]]; then
+          echo "No previous tag found ${{ steps.previous-tag.outputs.previous-tag }}, generating changelog for initial version only: v${{ steps.bump.outputs.new-version }}"
+          # For initial version, get all commits up to HEAD
+          generate-changelog -t HEAD
+        else
+          echo "Generating changelog from ${{ steps.previous-tag.outputs.previous-tag }} to ${{ steps.bump.outputs.new-version }}"
+          generate-changelog -t ${{ steps.previous-tag.outputs.previous-tag }}...${{ steps.bump.outputs.new-version }}
+        fi
+
+        # Remove existing "Changelog" header anywhere in the file
+        sed -i '/^# Changelog$/d' CHANGELOG.md
+        # Prepend the header to the top of the file
+        printf "# Changelog\n\n" | cat - CHANGELOG.md > temp && mv temp CHANGELOG.md
+        # Format headers remove dates and unreleased
+        sed -i -E 's/^(#+ [0-9]+\.[0-9]+\.[0-9]+) \(([Uu]nreleased|[0-9]{4}-[0-9]{2}-[0-9]{2})\)/\1/' CHANGELOG.md
+
+      env:
+        GITHUB_TOKEN: ${{ steps.generate-github-token.outputs.token }}
+
+    - name: Commit changes
+      if: always()  # Ensure it runs even if previous steps are skipped
+      shell: bash
+      run: |
+        git add package.json
+        git add CHANGELOG.md || true  # No-op if changelog not generated
+        git commit -m "Bump version to ${{ steps.bump.outputs.new-version }}"
+        git push origin main
+      env:
+        GITHUB_TOKEN: ${{ steps.generate-github-token.outputs.token }}
+
+    - name: Push latest version
+      shell: bash
+      run: git push origin main
```

Comment:
Do we need this? We already did `git push origin main` at the end of the previous step

Reply:
Removed, good call!